src/modules/FairFindPackage2.cmake.in

```diff
@@ -198,6 +210,89 @@ macro(find_package2 qualifier pkgname)
     find_package(${pkgname} ${__version__} QUIET ${ARGS_UNPARSED_ARGUMENTS})
   endif()
 
+  if(NOT DEFINED ${pkgname}_PREFIX AND ${pkgname}_FOUND AND NOT FAIR_NO_AUTO_DETECT_PREFIX)
+    # try to detect install prefix
+    set(__candidates)
+    set(__subdir_candidates)
+    foreach(__pkg IN ITEMS ${pkgname} ${pkgname_upper} ${pkgname_lower})
+      # guess targets
+      foreach(__pkg2 IN ITEMS ${pkgname} ${pkgname_upper} ${pkgname_lower})
+        foreach(__tgt IN ITEMS
+          "${__pkg}"
+          "${__pkg}::headers"
+          "${__pkg}::${__pkg2}"
+          "${__pkg}::${__pkg2}_shared"
+          "${__pkg}::${__pkg2}_static"
+        )
+          if(TARGET ${__tgt})
+            get_target_property(__includes ${__tgt} INTERFACE_INCLUDE_DIRECTORIES)
+            if(__includes)
+              list(APPEND __subdir_candidates "${__includes}")
+            endif()
+            unset(__includes)
+            get_target_property(__location ${__tgt} INTERFACE_LOCATION)
+            if(__location)
+              list(APPEND __subdir_candidates "${__location}")
+            endif()
+            unset(__location)
+          endif()
+        endforeach()
+      endforeach()
+
+      # guess variables
+      foreach(__var IN ITEMS
+        ${${__pkg}_BIN_DIR}
+        ${${__pkg}_BINDIR}
+        ${${__pkg}_INCLUDE_DIR}
+        ${${__pkg}_INCLUDEDIR}
+        ${${__pkg}_INCLUDE_DIRS}
+        ${${__pkg}_INCLUDEDIRS}
+        ${${__pkg}_INCDIR}
+        ${${__pkg}_LIBRARY_DIR}
+        ${${__pkg}_LIBRARYDIR}
+        ${${__pkg}_LIBRARY_DIRS}
+        ${${__pkg}_LIBRARYDIRS}
+        ${${__pkg}_LIBRARY_SHARED}
+        ${${__pkg}_LIBRARY_STATIC}
+        ${${__pkg}_LIBDIR}
+        ${${__pkg}_DIR}
+      )
+        if(__var)
+          list(APPEND __subdir_candidates ${__var})
+        endif()
+      endforeach()
+      list(APPEND __candidates
+        ${${__pkg}_EXECUTABLE}
+        ${${__pkg}_PREFIX}
+        ${${__pkg}_INSTALL_PREFIX}
+        ${${__pkg}_INSTALLPREFIX}
+        ${${__pkg}_INSTALL}
+      )
+    endforeach()
+    list(APPEND __candidates ${PACKAGE_PREFIX_DIR})
+
+    foreach(__subcand IN LISTS __subdir_candidates)
+      # we assume that these paths are least one directory
+      # level below the prefix
+      if(__subcand)
+        get_filename_component(__cand ${__subcand}/.. ABSOLUTE)
+        list(APPEND __candidates "${__cand}")
+        unset(__cand)
+      endif()
+    endforeach()
+    if(__candidates)
+      list(REMOVE_DUPLICATES __candidates)
+      __find_package2_lcd(OUTVAR ${pkgname}_PREFIX STRINGS ${__candidates})
+      message(VERBOSE "Guessing '${pkgname}' install prefix from: '${__candidates}' --> '${${pkgname}_PREFIX}'")
+    endif()
+    unset(__subdir_candidates)
+    unset(__candidates)
+  endif()
+
+  if(${pkgname}_PREFIX)
+    set(${pkgname}_PREFIX "${${pkgname}_PREFIX}" CACHE PATH "Auto-detected install prefix of package '${pkgname}'")
+  endif()
```

Comment:
Why not inside the `if(NOT DEFINED ${pkgname}_PREFIX`?

Reply:
done